.travis.yml

```diff
@@ -1,12 +1,178 @@
-language: c
-compiler:
-  - gcc
-before_install:
- - sudo apt-get update -qq
- - sudo apt-get install autotools-dev automake autoconf libtool libdbus-1-dev libcap-dev libblkid-dev libmount-dev libpam-dev libcryptsetup-dev libaudit-dev libacl1-dev libattr1-dev libselinux-dev liblzma-dev libgcrypt-dev libqrencode-dev libmicrohttpd-dev gperf python2.7-dev
-script: ./autogen.sh && ./configure && make V=1 && sudo ./systemd-machine-id-setup && make check && make distcheck
-after_failure: cat test-suite.log
+sudo: required
+
+services:
+    - docker
+
+jobs:
+    include:
+        - stage: build docker image
+          env:
+              # The machine id will be passed to Dockerfile for later checks
+              - MACHINE_ID=$(cat /var/lib/dbus/machine-id)
+          before_script: &update
+              # Ensure the latest version of docker is installed
+              - sudo apt-get update
+              - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
+              - docker --version
+              - env > .env
+          script:
+                # Copy content of CI_DIR into WORKDIR
+              - find $CI_DIR -maxdepth 1 -type f -exec cp -t . {} +
+              - echo "ENV GIT_SHA ${TRAVIS_COMMIT}" >> Dockerfile
+              - echo "ENV MACHINE_ID ${MACHINE_ID}" >> Dockerfile
+              - echo "$(git log -1 ${TRAVIS_COMMIT})" >> COMMITINFO
+                # Build docker container
+              - $CI_SCRIPT_DIR/build-docker-image.sh
+
+              - docker login -u="${DOCKER_USERNAME}" -p="${DOCKER_PASSWORD}"
+              - docker push ${DOCKER_REPOSITORY}
+
+        - stage: build
+          language: c
+          compiler: gcc
+          env:
+              # The machine id will be passed to container
+              - MACHINE_ID=$(cat /var/lib/dbus/machine-id)
```

Comment:
Why should the container have the same machine ID as the host? And /etc/machine-id would be more apppriate, unless the host is too old to have it. But this doesn't matter much either way, so let's keep it as is.

Reply:
>Why should the container have the same machine ID as the host?

There is basically no reason for the container to have the same machine ID as the host, it seemed easily to set it up this way instead of placing something totally random, though.

> /etc/machine-id would be more apppriate

I know, I'd done some research about that before I implemented this and it seems that the Travis-CI uses the /var/lib/dbus/machine-id. See for example https://github.com/travis-ci/packer-templates/issues/520